static/merge.js

```diff
@@ -0,0 +1,325 @@
+        const constructId = construct.split('-')[1];
+        const levelName = $(this).data('level-name');
+        const levelCount = $(this).data('levels-count');
+
+        // No reason to merge when only one sublevel is used.
+        if (sublevelsAmount > 1) {
+            // Initialize mergedHorizontalLevels with level ID which is getting merge.
+            mergedHorizontalLevels[levelId] = {};
+
+            // Get all headers for level.
+            var headers = $(`.star_chart_table-${constructId} tr:eq(1)`)
+                .find(`[data-level-id='${levelId}']`);
+
+            // Check if merged sublevels are inside last level in table.
+            isLast = isLastLevel(constructId, construct, headers, levelCount);
+
+            // Add cells which are inside sublevels columns to `stars` array then remove
+            // those columns and ca;culate `startIndex`.
+            // In addition each sublevel and his observations
+            // are added to mergedHorizontalLevels dictionary.
+            [...Array(headers.length).keys()].map(i => {
+                mergedHorizontalLevels[levelId][i] = {};
+                mergedHorizontalLevels[levelId][i][0] = getSublevelData(headers[i])
+                stars[i] = removeColumns(headers, i, constructId, construct);
+            });
+
+            // Initialize dictionary with amout of observations per cell.
+            var star_amount = {};
+
+            // Get first <th> element for specific table.
+            var header = $(`.star_chart_table-${constructId} tr:eq(1)`).first();
+
+            // If merged level is last inside table I have to add new header after the last element which left.
+            // Otherwise add new header before sublevel which was first inside table before removing it.
+            isLast 
+                ? $(header).find('th').eq(startIndex - 1)
+                    .after(`<th style="text-align:center;" class="align-middle" ` +
+                        `scope="col" data-toggle="tooltip" data-level-id="${levelId}">${levelName}</th>`)
+                : $(header).find('th').eq(startIndex)
+                    .before(`<th style="text-align:center;" class="align-middle" ` +
+                        `scope="col" data-toggle="tooltip" data-level-id="${levelId}">${levelName}</th>`);
+
+            // Calculate observations inside cells.
+            for (var i = 0; i < stars[0].length; i++) {
+                star_amount[i] = []
+
+                for (var j = 0; j < stars.length; j++) {
+                    const observations = $(stars[j][i]).data('modal-launch-observations');
+                    
+                    if (observations) {
+                        // Add observations and filter reoccurring ones.
+                        observations.map(observation => {
+                            if (star_amount[i].filter(star => observation === star).length === 0) {
+                                star_amount[i].push(observation);
```

Comment:
Is this like 7/8 level of indentation. That's another argument why we should devide this one.